patchwork-registries/src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -58,4 +213,17 @@ public static Identifier checkPrefix(String name, boolean warnOverrides) {
 
 		return new Identifier(prefix, newName);
 	}
+
+	@SuppressWarnings("unchecked")
+	public static <V extends IForgeRegistryEntry<V>> IForgeRegistry<V> wrapVanilla(Identifier identifier, Registry<?> registry) {
+		RegistryBuilder<V> builder = (RegistryBuilder<V>) GameData.getBuilder(identifier);
+
+		if (builder.getType().isAssignableFrom(StructureFeature.class)) {
+			// TODO: StructureFeature Needs slave map impl
+			return null;
+		}
+
+		builder.setVanillaRegistry((Registry<V>) registry);
```

Comment:
yes, this can happen.
I add a check in the latest commit so it wont crash.
However we should think about making Fabric registry visible to forge mods